### PR TITLE
Add Recommended Mark Names section.

### PR DESCRIPTION
**Pull request by anniesullie**
This addresses #103.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/user-timing/pull/106.html" title="Last updated on Nov 6, 2023, 7:49 PM UTC (3a3e0df)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/user-timing/106/4d5bae1...3a3e0df.html" title="Last updated on Nov 6, 2023, 7:49 PM UTC (3a3e0df)">Diff</a>

### CODE REVIEW

**Comment by anniesullie** [comment]
Note: this pulls from the L1 spec: https://www.w3.org/TR/user-timing-1/#:~:text=interface%20%5BNavigation%20Timing%5D.-,recommended%20mark%20names,-Developers%20are%20encouraged

File: index.html [hunk 1]
```diff
@@ -410,6 +410,26 @@ <h2>Privacy and Security</h2>
     mark or measure recorded by any script included on the page can be read by
     any other script running on the same page, regardless of origin.</p>
   </section>
+  <section id="recommended-mark-names" class="informative">
```
Comment (clelland) [inline index.html:413]: Quick drive-by comment: I think I'd move this up in the document, before the Privacy and Security sections, which are traditionally right before the appendices.

File: index.html [hunk 2]
```diff
@@ -410,6 +410,26 @@ <h2>Privacy and Security</h2>
...
+    <h2>Recommended Mark Names</h2>
+    <p>Developers are encouraged to use the following Recommended Mark Names to
+    mark common timings. The user agent is responsible for storing a new
+    timestamp under the specified mark name for Recommended Mark Names, just
+    like any user specified mark name. The user agent does not validate that the
```
Comment (clelland) [inline index.html:418]: Should this be "user-specified"?

**Review by clelland — COMMENTED** [review]
This looks good, in a drive-by review :)

File: index.html (hunk 1 above)
Comment (anniesullie) [reply index.html]: Updated.

File: index.html (hunk 2 above)
Comment (anniesullie) [reply index.html]: Oops, fixed.

File: index.html [hunk 3]
```diff
@@ -395,6 +395,25 @@ <h2>Convert a <var>name</var> to a <var>timestamp</var></h2>
         The <a data-cite="NAVIGATION-TIMING#performancetiming">PerformanceTiming</a> interface was defined in [[NAVIGATION-TIMING]] and is now considered obsolete. The use of names from the <a data-cite="NAVIGATION-TIMING#performancetiming">PerformanceTiming</a> interface is supported to remain backwards compatible, but there are no plans to extend this functionality to names in the <a data-cite="NAVIGATION-TIMING-2#dom-performancenavigationtiming">PerformanceNavigationTiming</a> interface defined in [[NAVIGATION-TIMING-2]] (or other interfaces) in the future.
       </p>
     </section>
+  </section><section id="recommended-mark-names" class="informative">
+    <h2>Recommended Mark Names</h2>
+    <p>Developers are encouraged to use the following Recommended Mark Names to
+      mark common timings. The user agent is responsible for storing a new
```
Comment (yoavweiss) [inline index.html:401]: I think we can drop the "the user agent is responsible..." sentence, as it doesn't add any information.

File: index.html [hunk 4]
```diff
@@ -395,6 +395,25 @@ <h2>Convert a <var>name</var> to a <var>timestamp</var></h2>
...
+      timestamp under the specified mark name for Recommended Mark Names, just
+      like any user-specified mark name. The user agent does not validate that the
+      usage of the Recommended Mark Name is appropriate or consistent with its
+      description.</p>
```
Comment (yoavweiss) [inline index.html:405]: I think it's worthwhile to add a not indicating why we recommend people do this. Maybe something like:
```<p class=note>Adding such Recommended Mark Names can help RUM providers and browsers to collect web developer signals regarding their application's performance at scale, and surface this information to developers without requiring any site-specific work</p>``` (or something to that effect)

File: index.html [hunk 5]
```diff
@@ -395,6 +395,25 @@ <h2>Convert a <var>name</var> to a <var>timestamp</var></h2>
         The <a data-cite="NAVIGATION-TIMING#performancetiming">PerformanceTiming</a> interface was defined in [[NAVIGATION-TIMING]] and is now considered obsolete. The use of names from the <a data-cite="NAVIGATION-TIMING#performancetiming">PerformanceTiming</a> interface is supported to remain backwards compatible, but there are no plans to extend this functionality to names in the <a data-cite="NAVIGATION-TIMING-2#dom-performancenavigationtiming">PerformanceNavigationTiming</a> interface defined in [[NAVIGATION-TIMING-2]] (or other interfaces) in the future.
       </p>
     </section>
+  </section><section id="recommended-mark-names" class="informative">
+    <h2>Recommended Mark Names</h2>
+    <p>Developers are encouraged to use the following Recommended Mark Names to
+      mark common timings. The user agent is responsible for storing a new
+      timestamp under the specified mark name for Recommended Mark Names, just
```
Comment (yoavweiss) [inline index.html:402]: I don't know that we need to capitalize that "Recommended Mark Names". If we want this to be a concept, it might be worthwhile to `<dfn>` it once, and then link to it.

File: index.html (hunk 4 above)
Comment (anniesullie) [reply index.html]: Good point! Done.

File: index.html (hunk 5 above)
Comment (anniesullie) [reply index.html]: Remove capitalization.

File: index.html [hunk 6]
```diff
@@ -395,6 +395,28 @@ <h2>Convert a <var>name</var> to a <var>timestamp</var></h2>
         The <a data-cite="NAVIGATION-TIMING#performancetiming">PerformanceTiming</a> interface was defined in [[NAVIGATION-TIMING]] and is now considered obsolete. The use of names from the <a data-cite="NAVIGATION-TIMING#performancetiming">PerformanceTiming</a> interface is supported to remain backwards compatible, but there are no plans to extend this functionality to names in the <a data-cite="NAVIGATION-TIMING-2#dom-performancenavigationtiming">PerformanceNavigationTiming</a> interface defined in [[NAVIGATION-TIMING-2]] (or other interfaces) in the future.
       </p>
     </section>
+  </section><section id="recommended-mark-names" class="informative">
+    <h2>Recommended mark names</h2>
```
Comment (npm1) [inline index.html:399]: Inconsistent casing

File: index.html [hunk 7]
```diff
@@ -395,6 +395,28 @@ <h2>Convert a <var>name</var> to a <var>timestamp</var></h2>
...
+    <p>Developers are encouraged to use the following recommended mark names to
+    mark common timings. The user agent does not validate that the usage of
+    the recommended mark name is appropriate or consistent with its
```
Comment (npm1) [inline index.html:402]: Overly repetitive text

File: index.html [hunk 8]
```diff
@@ -395,6 +395,28 @@ <h2>Convert a <var>name</var> to a <var>timestamp</var></h2>
         The <a data-cite="NAVIGATION-TIMING#performancetiming">PerformanceTiming</a> interface was defined in [[NAVIGATION-TIMING]] and is now considered obsolete. The use of names from the <a data-cite="NAVIGATION-TIMING#performancetiming">PerformanceTiming</a> interface is supported to remain backwards compatible, but there are no plans to extend this functionality to names in the <a data-cite="NAVIGATION-TIMING-2#dom-performancenavigationtiming">PerformanceNavigationTiming</a> interface defined in [[NAVIGATION-TIMING-2]] (or other interfaces) in the future.
       </p>
     </section>
+  </section><section id="recommended-mark-names" class="informative">
```
Comment (npm1) [inline index.html:398]: I'm not sure this can be non-normative since it is adding a bunch of definitions

File: index.html [hunk 9]
```diff
@@ -395,6 +395,28 @@ <h2>Convert a <var>name</var> to a <var>timestamp</var></h2>
...
+    <h2>Recommended mark names</h2>
+    <p>Developers are encouraged to use the following recommended mark names to
+    mark common timings. The user agent does not validate that the usage of
+    the recommended mark name is appropriate or consistent with its
+    description.</p>
+    <p class="note">Adding such recommended mark names can help performance
+    tools tailor guidance to a site. These mark names can also help real user
+    monitoring providers and browsers collect web developer signals regarding
```
Comment (npm1) [inline index.html:406]: user agents

File: index.html [hunk 10]
```diff
@@ -395,6 +395,28 @@ <h2>Convert a <var>name</var> to a <var>timestamp</var></h2>
...
+    their application's performance at scale, and surface this information to
+    developers without requiring any site-specific work.</p>
+    <dl>
+      <dt>"mark_fully_loaded"</dt>
```
Comment (npm1) [inline index.html:410]: These seem definitions to me.

File: index.html [hunk 11]
```diff
@@ -395,6 +395,28 @@ <h2>Convert a <var>name</var> to a <var>timestamp</var></h2>
...
+      <dd>The time when the page is considered fully loaded as marked by the
+        developer in their application.</dd>
+      <dt>"mark_fully_visible"</dt>
+      <dd>The time when the page is considered completely visible to an end-user
```
Comment (npm1) [inline index.html:414]: fully visible? Why is inconsistent wording used

File: index.html [hunk 12]
```diff
@@ -395,6 +395,28 @@ <h2>Convert a <var>name</var> to a <var>timestamp</var></h2>
...
+        as marked by the developer in their application.</dd>
+      <dt>"mark_interactive"</dt>
+      <dd>The time when the page is considered fully interactive to an end-user
```
Comment (npm1) [inline index.html:417]: fully interactive vs interactive

**Review by npm1 — CHANGES_REQUESTED** [review]

File: index.html (hunk 8 above)
Comment (yoavweiss) [reply index.html:398]: Yeah, we can remove the class=informative bit

File: index.html (hunk 10 above)
Comment (yoavweiss) [reply index.html:410]: Yeah. We should probably use [<dfn>](https://respec.org/docs/#definitions-and-linking) to define these.

File: index.html (hunk 6 above)
Comment (yoavweiss) [reply index.html:399]: Inconsistent with what? :)

File: index.html [hunk 13]
```diff
@@ -395,6 +395,28 @@ <h2>Convert a <var>name</var> to a <var>timestamp</var></h2>
         The <a data-cite="NAVIGATION-TIMING#performancetiming">PerformanceTiming</a> interface was defined in [[NAVIGATION-TIMING]] and is now considered obsolete. The use of names from the <a data-cite="NAVIGATION-TIMING#performancetiming">PerformanceTiming</a> interface is supported to remain backwards compatible, but there are no plans to extend this functionality to names in the <a data-cite="NAVIGATION-TIMING-2#dom-performancenavigationtiming">PerformanceNavigationTiming</a> interface defined in [[NAVIGATION-TIMING-2]] (or other interfaces) in the future.
       </p>
     </section>
+  </section><section id="recommended-mark-names" class="informative">
+    <h2>Recommended mark names</h2>
+    <p>Developers are encouraged to use the following recommended mark names to
```
Comment (yoavweiss) [inline index.html:400]: Maybe "recommended mark names" here should be a <dfn>?

Comment (anniesullie) [reply index.html:400]: I think you were trying to write an HTML tag and github ate everything between the angle brackets?

Should be what?

File: index.html (hunk 10 above)
Comment (anniesullie) [reply index.html:409]: I think GitHub ate an HTML tag here, @yoavweiss. What should we use to define these?

File: index.html (hunk 13 above)
Comment (anniesullie) [reply index.html:400]: I added `<dfn>` around these, but now I get an error:

> Found definition for "Recommended mark names", but nothing links to it. This is usually a spec bug!

Comment (jyasskin) [reply index.html:400]: Use `data-export` to declare that you expect the definition to be used from outside this spec. I actually wouldn't `<dfn>` this header, but `<dfn>`'ing the terms is good.

File: index.html [hunk 14]
```diff
@@ -395,6 +395,27 @@ <h2>Convert a <var>name</var> to a <var>timestamp</var></h2>
         The <a data-cite="NAVIGATION-TIMING#performancetiming">PerformanceTiming</a> interface was defined in [[NAVIGATION-TIMING]] and is now considered obsolete. The use of names from the <a data-cite="NAVIGATION-TIMING#performancetiming">PerformanceTiming</a> interface is supported to remain backwards compatible, but there are no plans to extend this functionality to names in the <a data-cite="NAVIGATION-TIMING-2#dom-performancenavigationtiming">PerformanceNavigationTiming</a> interface defined in [[NAVIGATION-TIMING-2]] (or other interfaces) in the future.
       </p>
     </section>
+  </section><section id="recommended-mark-names">
+    <h2><dfn>Recommended mark names</dfn></h2>
```
Comment (marcoscaceres) [inline index.html:399]: Unfortunately, ReSpec doesn't yet support data-export on headers, so <dfn class="export"> is good: 

```suggestion
    <h2><dfn class="export">Recommended mark names</dfn></h2>
```

File: index.html (hunk 13 above)
Comment (marcoscaceres) [reply index.html:400]: (see suggestion above...)

File: index.html [hunk 15]
```diff
@@ -395,6 +395,27 @@ <h2>Convert a <var>name</var> to a <var>timestamp</var></h2>
...
+      <dt><dfn>"mark_fully_loaded"</dfn></dt>
+      <dd>The time when the page is considered fully loaded as marked by the
+        developer in their application.</dd>
+      <dt><dfn>"mark_fully_visible"</dfn></dt>
+      <dd>The time when the page is considered fully visible to an end-user
+        as marked by the developer in their application.</dd>
+      <dt><dfn>"mark_interactive"</dfn></dt>
+      <dd>The time when the page is considered interactive to an end-user as
+        marked by the developer in their application.</dd>
+    </dl>
```
Comment (marcoscaceres) [inline index.html:418]: Might be good to add an example.

Comment (anniesullie) [reply index.html:418]: Great idea! Added one.

File: index.html (hunk 6 above)
Comment (anniesullie) [reply index.html:399]: @npm1 can you clarify?

Comment (npm1) [reply index.html:399]: With other headers. But it looks like spec is not consistent anyways

File: index.html [hunk 16]
```diff
@@ -395,6 +395,42 @@ <h2>Convert a <var>name</var> to a <var>timestamp</var></h2>
...
+      <dt><dfn class="export">"mark_fully_visible"</dfn></dt>
+      <dd>The time when the page is considered fully visible to an end-user
+        as marked by the developer in their application.</dd>
+      <dt><dfn class="export">"mark_interactive"</dfn></dt>
+      <dd>The time when the page is considered interactive to an end-user as
+        marked by the developer in their application.</dd>
+    </dl>
+    <p>In this example, the page asynchonously initializes a chat widget, a
+    searchbox, and a newsfeed upon loading. When finished, the
+    mark_fully_loaded mark name enables lab tools and analytics providers to
```
Comment (npm1) [inline index.html:421]: Use the dfn?

File: index.html [hunk 17]
```diff
@@ -395,6 +395,42 @@ <h2>Convert a <var>name</var> to a <var>timestamp</var></h2>
         The <a data-cite="NAVIGATION-TIMING#performancetiming">PerformanceTiming</a> interface was defined in [[NAVIGATION-TIMING]] and is now considered obsolete. The use of names from the <a data-cite="NAVIGATION-TIMING#performancetiming">PerformanceTiming</a> interface is supported to remain backwards compatible, but there are no plans to extend this functionality to names in the <a data-cite="NAVIGATION-TIMING-2#dom-performancenavigationtiming">PerformanceNavigationTiming</a> interface defined in [[NAVIGATION-TIMING-2]] (or other interfaces) in the future.
       </p>
     </section>
+  </section><section id="recommended-mark-names">
+    <h2>Recommended mark names</h2>
+    <p>Developers are encouraged to use the following recommended mark names to
+    mark common timings. The user agent does not validate that the usage of
```
Comment (yoavweiss) [inline index.html:401]: "user agent" should link to https://infra.spec.whatwg.org/#user-agent

File: index.html [hunk 18]
```diff
@@ -395,6 +395,42 @@ <h2>Convert a <var>name</var> to a <var>timestamp</var></h2>
...
+      <dd>The time when the page is considered fully loaded as marked by the
+        developer in their application.</dd>
+      <dt>"<dfn class="export">mark_fully_visible</dfn>"</dt>
+      <dd>The time when the page is considered fully visible to an end-user
+        as marked by the developer in their application.</dd>
+      <dt>"<dfn class="export">mark_interactive</dfn>"</dt>
+      <dd>The time when the page is considered interactive to an end-user as
+        marked by the developer in their application.</dd>
+    </dl>
+    <p>In this example, the page asynchonously initializes a chat widget, a
```
Comment (yoavweiss) [inline index.html:419]: Can you surround this with a note div?

**Review by yoavweiss — COMMENTED** [review]
Looks great!  A couple of nits

**Review by yoavweiss — APPROVED** [review]
LGTM

**Comment by yoavweiss** [comment]
@mmocny suggested offline that it might be interesting to think through how should developers report these timings. That can be as a followup.

**Review by npm1 — COMMENTED** [review]
Should probably be reviewed by someone outside of Chrome, otherwise this PR is not useful.

**Comment by anniesullie** [comment]
> Should probably be reviewed by someone outside of Chrome, otherwise this PR is not useful.

@nicjansma do you have time to review?

**Comment by nicjansma** [comment]
LGTM as well!

**Review by nicjansma — APPROVED** [review]

File: index.html [hunk 19]
```diff
@@ -395,6 +395,44 @@ <h2>Convert a <var>name</var> to a <var>timestamp</var></h2>
         The <a data-cite="NAVIGATION-TIMING#performancetiming">PerformanceTiming</a> interface was defined in [[NAVIGATION-TIMING]] and is now considered obsolete. The use of names from the <a data-cite="NAVIGATION-TIMING#performancetiming">PerformanceTiming</a> interface is supported to remain backwards compatible, but there are no plans to extend this functionality to names in the <a data-cite="NAVIGATION-TIMING-2#dom-performancenavigationtiming">PerformanceNavigationTiming</a> interface defined in [[NAVIGATION-TIMING-2]] (or other interfaces) in the future.
       </p>
     </section>
+  </section><section id="recommended-mark-names">
```
Comment (npm1) [inline index.html:398]: `</section>` better in its own line...

**Review by npm1 — APPROVED** [review]
